homeassistant/components/template/light.py

```diff
@@ -321,12 +298,12 @@ def effect_list(self) -> list[str] | None:
         return self._effect_list
 
     @property
-    def color_mode(self):
+    def color_mode(self) -> ColorMode | None:
         """Return current color mode."""
         return self._color_mode
 
     @property
-    def supported_color_modes(self):
+    def supported_color_modes(self) -> set[ColorMode] | set[str] | None:
```

Comment:
I think the `set[str]` is just for backwards compatibility in the base class
```suggestion
    def supported_color_modes(self) -> set[ColorMode] | None:
```